### PR TITLE
[MRG+1] Example: remove overlapping text from image_masked.py

### PR DESCRIPTION
**Pull request by efiring**
This also makes several other changes that clarify the example
and improve the code.

### CODE REVIEW

**Comment by efiring** [comment]
While looking for examples using a proportional colorbar, I noticed that the image_masked.py example looks terrible, with colorbar tick labels smashed together and axes titles overflowing their bounds.  In this PR I spruce it up with some rearrangement, and edit and add labels to make it a bit easier to see what the example is showing.  It's still not great, but it's better--no longer such a gross embarrassment.
image_masked

File: examples/pylab_examples/image_masked.py [hunk 1]
```diff
@@ -12,8 +12,10 @@
 import numpy as np
 
 # compute some interesting data
-delta = 0.025
-x = y = np.arange(-3.0, 3.0, delta)
+x0, x1 = -5, 5
+y0, y1 = -3, 3
+x = np.linspace(x0, x1, 500)
```
Comment (NelleV) [inline examples/pylab_examples/image_masked.py]: :+1:

File: examples/pylab_examples/image_masked.py [hunk 2]
```diff
@@ -38,24 +40,33 @@
 # Anything above that range is colored based on palette.set_over, etc.
 
 # set up the axes
-fig, (ax1, ax2) = plt.subplots(1, 2)
+fig, (ax1, ax2) = plt.subplots(2, 1, figsize=(6, 5.4))
```
Comment (NelleV) [inline examples/pylab_examples/image_masked.py]: Isn't the second argument unecessary?
I also tend to favor being explicit : `nrows=2`.

File: examples/pylab_examples/image_masked.py [hunk 3]
```diff
@@ -38,24 +40,33 @@
...
-ax1.set_title('Green=low, Red=high, Blue=bad')
-fig.colorbar(im, extend='both', orientation='horizontal', shrink=0.8, ax=ax1)
+                norm=colors.Normalize(vmin=-1.0, vmax=1.0),
+                aspect='auto',
+                origin='lower',
+                extent=[x0, x1, y0, y1])
+ax1.set_title('Green=low, Red=high, Blue=masked')
+cbar = fig.colorbar(im, extend='both', shrink=0.9, ax=ax1)
+cbar.set_label('uniform')
+for ticklabel in ax1.xaxis.get_ticklabels():
```
Comment (NelleV) [inline examples/pylab_examples/image_masked.py]: The x-axis could be shared, which renders these two lines unecessary.

Comment (NelleV) [inline examples/pylab_examples/image_masked.py]: Actually, both axis could be shared.

**Comment by NelleV** [comment]
It would also be nice to use the new color scheme instead of the "r", "b", "g".

File: examples/pylab_examples/image_masked.py (hunk 2 above)
Comment (efiring) [inline examples/pylab_examples/image_masked.py]: Agreed.  I will fix this.

**Comment by efiring** [comment]
Regarding colors: In this case, I think that using the simple primary colors, with their simple name abbreviations in the code, is preferable for the purposes of illustration.

**Pull request merged**